js/source/image_source.js

```diff
@@ -57,19 +56,27 @@ ImageSource.prototype = util.inherit(Evented, {
     onAdd: function(map) {
         this.map = map;
         if (this.image) {
-            this.createTile();
+            this.setCoordinates();
         }
     },
 
     /**
-     * Calculate which mercator tile is suitable for rendering the image in
-     * and create a buffer with the corner coordinates. These coordinates
-     * may be outside the tile, because raster tiles aren't clipped when rendering.
-     * @private
+     * Update image coordinates and rerender map
+     *
+     * @param {Array} coordinates lng, lat coordinates in order clockwise
```

Comment:
Hard to argue against! I honestly only copied the argument descriptor from the constructor which constituted the existing public API so to speak. So one could argue that this should already have been present on the constructor.. but I will add it to both!